pkg/obfuscator/domain.go

```diff
@@ -54,13 +58,18 @@ func (d *domainObfuscator) obfuscatedDomain(domain string) string {
 	if replacement, ok := d.domainMapping[domain]; ok {
```

Comment:
the whole thing suffers from race conditions now because the domainMapping is not locked: it can happen that one thread has a domain mapping replaced with another mapping.

we have to do the same generateIfAbsent that we do in the other obfuscators too. As mentioned below, it makes sense to just refactor what we already did in MAC+IP into dedicated and correct logical structure and then reuse it here.

Reply:
actually, it suffered from races before as well :(